packages/gatsby/src/utils/websocket-manager.js

```diff
@@ -190,13 +227,19 @@ class WebsocketManager {
     this.staticQueryResults.set(data.id, data)
     if (this.isInitialised) {
       this.websocket.send({ type: `staticQueryResult`, payload: data })
+      telemetry.trackCli(`WEBSOCKET_EMIT_STATIC_PAGE_DATA_UPDATE`, {
+        siteMeasurements: { clientsCount: this.connectedClients },
+      })
     }
   }
 
   emitPageData(data: QueryResult) {
     this.pageResults.set(data.id, data)
     if (this.isInitialised) {
       this.websocket.send({ type: `pageQueryResult`, payload: data })
+      telemetry.trackCli(`WEBSOCKET_EMIT_PAGE_DATA_UPDATE`, {
+        siteMeasurements: { clientsCount: this.connectedClients },
+      })
```

Comment:
This might lead to unexpected results. If data update would make multiple queries/pages "dirty" (i.e. updating blog post cause updating blog posts listings, blog author personal page and updating actual blog post page) - this will be called N times for single data update

Reply:
yeah, I'd aggregate what all the pages are that are updated in a go and send that all together. That + what page they're looking at seems like the right "event" not updating each page.